File: Leetcode/Java/Medium/2958LengthofLongestSubarrayWithatMostKFrequency.java
class Solution {
    public int maxSubarrayLength(int[] nums, int k) {
        int left = 0;
        int max_length = 0;
        Map<Integer, Integer> count = new HashMap<>();

        for (int right = 0; right < nums.length; right++) {
            int num = nums[right];
            count.put(num, count.getOrDefault(num, 0) + 1);
            while(count.get(num) > k) {
                int leftNum = nums[left];
                count.put(leftNum, count.get(leftNum) -1);
                left++;
            }
            max_length = Math.max(max_length, right - left +1);
        }
        return max_length;
    }
}
//Accepted solution

File: Leetcode/Rust/Medium/2958LengthofLongestSubarrayWithatMostKFrequency.rs
impl Solution {
    pub fn max_subarray_length(nums: Vec<i32>, k: i32) -> i32 {
        let mut left = 0;
        let mut max_length = 0;
        let mut count = std::collections::HashMap::new();

        for right in 0..nums.len() {
            let num = nums[right];
            *count.entry(num).or_insert(0) += 1;
            while count[&num] > k {
                let left_num = nums[left];
                *count.get_mut(&left_num).unwrap() -= 1;
                left +=1;
            }
            max_length = max_length.max(right - left + 1);
        }
        max_length as i32
    }
}
//Accepted solution